pkg/cloud/services/networking/port_test.go

```diff
@@ -225,12 +229,68 @@ func Test_GetOrCreatePort(t *testing.T) {
 						ID: portID1,
 					}, nil)
 				m.ReplaceAllAttributesTags("ports", portID1, attributestags.ReplaceAllOpts{Tags: []string{"my-port-tag"}}).Return([]string{"my-port-tag"}, nil)
+				m.
+					ListSubnet(subnets.ListOpts{
+						Name:      "subnetFoo",
+						NetworkID: netID,
+					}).Return([]subnets.Subnet{
+					{
+						ID:        subnetID1,
+						Name:      "subnetFoo",
+						NetworkID: netID,
+					},
+				}, nil)
 			},
 			&ports.Port{
 				ID: portID1,
 			},
 			false,
 		},
+		{
+			"fails to create port with specified portOpts if subnet query returns more than one subnet",
```

Comment:
Maybe we want a similar test for if the networks query returns more than one network?

Reply:
I made a new suite to test this. Check the second commit